pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -269,10 +271,22 @@ private String getQueryResponse(String query, @Nullable SqlNode sqlNode, String
       return QueryException.getException(QueryException.QUERY_VALIDATION_ERROR, e).toString();
     }
     try {
-      String inputTableName =
-          sqlNode != null ? RequestUtils.getTableNames(CalciteSqlParser.compileSqlNodeToPinotQuery(sqlNode)).iterator()
-              .next() : CalciteSqlCompiler.compileToBrokerRequest(query).getQuerySource().getTableName();
-      tableName = _pinotHelixResourceManager.getActualTableName(inputTableName, database);
+      List<String> inputTableNames = new ArrayList<>();
+      if (sqlNode != null) {
+        if (sqlNode instanceof SqlShowTables) {
+          inputTableNames.addAll(_pinotHelixResourceManager.getAllTables(database));
```

Comment:
We don't need query access to all tables in order to run SHOW TABLES. Instead, we can take the same action as `PinotTableRestletResource.listTables()` which also list all tables.

Reply:
I've made this change. Can you please have a look?